assets/javascripts/discourse/components/discourse-post-event/index.gjs

```diff
@@ -0,0 +1,197 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import willDestroy from "@ember/render-modifiers/modifiers/will-destroy";
+import { service } from "@ember/service";
+import { htmlSafe } from "@ember/template";
+import concatClass from "discourse/helpers/concat-class";
+import routeAction from "discourse/helpers/route-action";
+import { emojiUnescape } from "discourse/lib/text";
+import { escapeExpression } from "discourse/lib/utilities";
+import i18n from "discourse-common/helpers/i18n";
+import Creator from "./creator";
+import Dates from "./dates";
+import Invitees from "./invitees";
+import MoreMenu from "./more-menu";
+import Status from "./status";
+import Url from "./url";
+
+export default class DiscoursePostEvent extends Component {
+  @service currentUser;
+  @service discoursePostEventApi;
+  @service messageBus;
+
+  @action
+  setupMessageBus() {
+    this.messageBus.subscribe(
+      "/discourse-post-event/" + this.args.event.post.topic.id,
+      async (msg) => {
+        const event = await this.discoursePostEventApi.event(msg.id);
+        this.args.event.updateFromEvent(event);
+      }
+    );
+  }
+
+  @action
+  teardownMessageBus() {
+    this.messageBus.unsubscribe(
+      "/discourse-post-event/" + this.args.event.post.topic.id
+    );
+  }
```

Comment:
I think @davidtaylorhq doesn't like doing this, especially when you are using args inside the body of the modifier, at least he would prefer I give it as argument of the modifier to avoid any recomputation (although it shouldn't happen here).

Reply:
Recomputation is often desired. did-insert and did-update are just a bridge for old patterns in glimmer world.

You can see how they're more similar to `@computed` and observers where you need to declare dependent properties, rather than being auto-updated whenever used props change.

We should come up with official dev-xp stance on this 😅

Reply:
I mean Im fine with the modifier, it's even what I did first :) I can change to this, and we can always alter later if David disagrees. :+1: